spec/lib/webauthn/authentication_service_spec.rb

```diff
@@ -241,4 +257,58 @@
       expect(COSE::Algorithm.find(alg)).not_to be_nil
     end
   end
+
+  describe "authenticating a valid passkey" do
+    let(:options) do
+      { factor_type: UserSecurityKey.factor_types[:first_factor], session: secure_session }
+    end
+
+    ##
+    # These are sourced from an actual key, see instructions at the top of this spec for details
+    #
+    let(:public_key) { valid_passkey_data[:public_key] }
+    let(:credential_id) { valid_passkey_data[:credential_id] }
+    let(:signature) { valid_passkey_auth_data[:signature] }
+    let(:authenticator_data) { valid_passkey_auth_data[:authenticatorData] }
+    let(:challenge) { valid_passkey_challenge }
+
+    let(:client_data_param) { passkey_client_data_param("webauthn.get") }
+
+    let!(:security_key) do
+      Fabricate(
+        :user_security_key,
+        credential_id: credential_id,
+        public_key: public_key,
+        user: security_key_user,
+        factor_type: UserSecurityKey.factor_types[:first_factor],
+        last_used: nil,
+        name: "A key",
+      )
+    end
+
+    it "works" do
+      key = service.authenticate_security_key
+      expect(key).to eq(security_key)
+      expect(key).to be_a(UserSecurityKey)
+      expect(key.user).to eq(current_user)
```

Comment:
This is also not really testing anything IMO because `key` has been determined to be equal to `security_key` and `security_key` was created with the `current_user`.